src/main/java/sk/hfa/projects/services/ProjectService.java

```diff
@@ -95,36 +108,26 @@ public void deleteById(Long id) {
         return result;
     }
 
-    @Override
-    public Project build(ProjectRequest request) {
-        if (Objects.isNull(request))
-            throw new InvalidProjectRequestException("Invalid request body");
-
-        if (!isValidCategory(request.getCategory()))
-            throw new IllegalArgumentException(INVALID_CATEGORY_MESSAGE);
-
-        Category projectCategory = getCategory(request.getCategory());
-        if (Category.COMMON.equals(projectCategory))
-            return CommonProject.build(request);
-        else if (Category.INDIVIDUAL.equals(projectCategory))
-            return IndividualProject.build(request);
-        else
-            return InteriorDesignProject.build(request);
-    }
-
-    private Category getCategory(String category) {
-        if (Category.COMMON.name().equals(category))
-            return Category.COMMON;
-        else if (Category.INDIVIDUAL.name().equals(category))
-            return Category.INDIVIDUAL;
-        else if (Category.INTERIOR_DESIGN.name().equals(category))
-            return Category.INTERIOR_DESIGN;
-
-        return null;
+    private Project build(ProjectRequest request, Category projectCategory) {
+        Image titleImage = imageService.save(request.getTitleImageFile());
+        List<Image> galleryImages = imageService.save(request.getGalleryImageFiles());
+
+        if (Category.COMMON.equals(projectCategory)) {
+            List<Image> floorPlanImages = imageService.save(((CommonProjectRequest) request).getFloorPlanImageFiles());
+            return CommonProject.build((CommonProjectRequest) request, titleImage, galleryImages, floorPlanImages);
+        } else if (Category.INDIVIDUAL.equals(projectCategory)) {
+            return IndividualProject.build((IndividualProjectRequest) request, titleImage, galleryImages);
+        } else {
+            return InteriorDesignProject.build((InteriorProjectRequest) request, titleImage, galleryImages);
+        }
     }
 
-    private boolean isValidCategory(String category) {
-        return !Objects.isNull(getCategory(category));
+    private void deleteImages(Project project) {
+        imageService.deleteImage(project.getTitleImage());
+        imageService.deleteImages(project.getGalleryImages());
+        if (Category.COMMON.equals(project.getCategory())) {
+            imageService.deleteImages(((CommonProject) Hibernate.unproxy(project)).getFloorPlanImages());
```

Comment:
Hibernate.unproxy(project, CommonProject.class);

Reply:
Načo to slúži?

Reply:
Done